slides/testing/parser/file-parser.ts

```diff
@@ -0,0 +1,16 @@
+import { File } from './file';
+import { IParser } from './parser.interface';
+
+export class FileParser implements IParser<Array<File>> {
```

Comment:
Is it worth not only creating a class, but creating and implementing an `IParser` interface for what is otherwise just a pure function?

This could simply be something like...
```ts
export function parseSlides({ content, fileName }): Array<File> {
  const slideIds = content.match(/<!-- .slide: id.*-->/g) || [];
  return slideIds.map(slideId => ({ fileName, slideId }));
}
```